source/administrator-guide/managing-users-and-groups/authentication-mode.rst

```diff
@@ -510,6 +511,291 @@ You can configure your environment by updating the previous file or by defining
+    #. Press "Create" and and add a "Claim to Role".  
+    #. Set Sync Mode Override to "Force"
+    #. Claim: `roles`
+    #. Claim Value: `name of the administrator role in the backing IDP`
+    #. Role: choose the "Administrator" role from the `myclient` client.
+    #. Repeat the above for Administrator, Editor, Reviewer, and RegisteredGuest
+
+#. Configure details for your backing IDP
+
+    #. Edit the "Identity Provider" you just configured
+    #. On the Mappers tab, "Add Builtin" and tick "client roles (User Client Role)" then "Add selected"
+    #. Edit the "client roles" mapper and make sure "Add to ID token" and "Add to userinfo" are on
+   
+
+
+You should have Keycloak's Client id ("myclient") and the client secret.  The configuration JSON is available at `https://YOUR_KEYCLOAK_HOST/realms/{YOUR REALM NAME}/.well-known/openid-configuration```
+
+Your environment variables will looks like this;
+
+.. code-block:: properties
+
+    GEONETWORK_SECURITY_TYPE=openidconnect
+    OPENIDCONNECT_CLIENTSECRET='...'
+    OPENIDCONNECT_CLIENTID='...'   
+    OPENIDCONNECT_SERVERMETADATA_JSON_TEXT='...big json text...' 
+    OPENIDCONNECT_IDTOKENROLELOCATION='resource_access.{your client id}.roles'
+
+
+Azure AD Configuration
+======================
+
+There are two ways to setup Azure AD.  The first is with user and groups (a more traditional LDAP method) or with Application Roles.
+
+With Users and Groups
+``````````````````````
+
+Setup the Azure Application;
+
+#. Create a new `App Registration`
+#. use "http://localhost:8080/geonetwork/login/oauth2/code/geonetwork-oicd" as a redirect URIs
+#. On the "Certificates & Secrets" add a new secret and record it (make sure you get the secret value and NOT the object id)
+#. Make sure the groups are in the ID token - on the "Manifest" tab, edit the JSON so that  "groupMembershipClaims": "SecurityGroup" is set
+#. On the summary page, get the Application (client) ID
+#. On the summary page, choose "Endpoints" (at the top) and get the JSON text from the "OpenID Connect metadata document" Endpoints
+
+Setup users and groups;
+
+#. In Azure AD, go to groups
+#. Add new Groups - "geonetworkAdmin", "geonetworkReviewer", etc...  Record the name and the group's **Object ID**
+#. Edit a User, and choose Groups, and add them to appropriate group.
+
+
+Your environment variables will looks like this;
+
+.. code-block:: properties
+
+    GEONETWORK_SECURITY_TYPE=openidconnect
+    OPENIDCONNECT_CLIENTSECRET='...'
+    OPENIDCONNECT_CLIENTID='...'   
+    OPENIDCONNECT_SERVERMETADATA_JSON_TEXT='...big json text...' 
+    OPENIDCONNECT_IDTOKENROLELOCATION='groups'
+    OPENIDCONNECT_ROLECONVERTER='3a94275f-7d53-4205-8d78-11f39e9ffa5a=Administrator,d93c6444-feee-4b67-8c0f-15d6796370cb=Reviewer'
+
+
+NOTE: The roles are in the "roles" part of the ID Token.
+
+
+NOTE: The OPENIDCONNECT_ROLECONVERTER converts the Azure AD Group's Object ID to a Geonetwork Role.
```

Comment:
Better to use Sphinx annotation `.. note::`

Reply:
changed